engine/src/main/java/com/google/android/fhir/sync/upload/UploaderImpl.kt

```diff
@@ -32,26 +32,36 @@ import org.hl7.fhir.r4.model.Resource
 import org.hl7.fhir.r4.model.ResourceType
 import timber.log.Timber
 
-/** [Uploader] implementation to work with Fhir [Bundle]. */
-internal class BundleUploader(
+/**
+ * Implementation of the [Uploader]. It orchestrates the pre processing of [LocalChange] and
+ * constructing appropriate upload requests via [UploadWorkManager] and uploading of requests via
+ * [DataSource]. [Uploader] clients should call upload and listen to the various states emitted by
+ * [UploadWorkManager] as [UploadResult].
+ */
+internal class UploaderImpl(
   private val dataSource: DataSource,
-  private val bundleGenerator: TransactionBundleGenerator,
-  private val localChangesPaginator: LocalChangesPaginator
+  private val uploadWorkManager: UploadWorkManager
```

Comment:
nit: add trailing comma at the end so when we go to add more parameters (hopefully never), we don't have to edit this line
```suggestion
  private val uploadWorkManager: UploadWorkManager,
```